PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -776,11 +758,13 @@ PayloadMain (
 #endif
 
   // Check if there is a pre-OS checker that needs to be executed
-  if (PldBase != NULL) {
-    mPreOsCheckerEntry = LoadPreOsChecker (PldBase);
+  Status = LocateComponentEntry (SIGNATURE_32 ('I', 'P', 'F', 'W'), SIGNATURE_32 ('P', 'O', 'S', 'C'), &ContainerEntry, &ComponentEntry);
+  if (!EFI_ERROR (Status)) {
+    mPreOsCheckerEntry = LoadPreOsChecker (
+      ContainerEntry->Base + ((CONTAINER_HDR *)ContainerEntry->HeaderCache)->DataOffset + ComponentEntry->Offset + sizeof (LOADER_COMPRESSED_HEADER)
```

Comment:
I am not sure this will be the best way to get the data of the 'POSC' binary data. When I try to something like was done here https://github.com/slimbootloader/slimbootloader/blob/master/BootloaderCorePkg/Stage2/Stage2.c#L188 it does not seem to be working for me in Os Loader.

Reply:
It is not the preferred way to load a component since it does not handle the verification and decompression.  It should just work with LoadCompoent() in OsLoader.   If not, we should treat it as a bug and need to be fixed.

Reply:
Retested using the LoadComponent( )routine and it is working, not sure what the issue was when testing previously.